walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -40,19 +40,20 @@
    * @return Collection<TimeRange> of available TimeRange's for the meeting to be scheduled
    */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    Collection<TimeRange> result;
+    Collection<TimeRange> result = new ArrayList<TimeRange>();
 
-    ArrayList<TimeRange> conflicts = generateConflicts(events, request.getAttendees());
+    ArrayList<TimeRange> mandatoryConflicts = generateConflicts(events, request.getAttendees());
+    ArrayList<TimeRange> optionalConflicts =
+        generateConflicts(events, request.getOptionalAttendees());
+    ArrayList<TimeRange> allConflicts = (ArrayList<TimeRange>) mandatoryConflicts.clone();
+    allConflicts.addAll(optionalConflicts);
 
-    if (conflicts.isEmpty()) {
-      result = new ArrayList<TimeRange>();
-      // if no conflicts exist, attempt to book the entire day
-      TimeRange fullDay =
-          TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true);
-      addIfPossible(result, fullDay, request.getDuration());
-    } else {
-      result = generateFreeTimeRanges(conflicts, request.getDuration());
+    result = generateFreeTimeRanges(allConflicts, request.getDuration());
+
+    if (result.isEmpty() && !request.getAttendees().isEmpty()) {
```

Comment:
so, you are trying to generate a result using optional attendees and their optional conflicts.

If nothing works out you loosen the requirements to just required attendees?

IMO a explanatory comment might be good here, either that or renaming some variables to be self explatory.

for ex: you can rename result to resultIncludingAllAttendees

then its clear what on 53 it means when  resultIncludingAllAttendees.isEmpty() means.